types/block.go

```diff
@@ -126,23 +128,100 @@ func (dah *DataAvailabilityHeader) Hash() []byte {
 	return dah.hash
 }
 
-func (dah *DataAvailabilityHeader) ToProto() (*tmproto.DataAvailabilityHeader, error) {
+// ValidateBasic runs stateless checks on the DataAvailabilityHeader. Calls Hash() if not already called
+func (dah *DataAvailabilityHeader) ValidateBasic() error {
 	if dah == nil {
-		return nil, errors.New("nil DataAvailabilityHeader")
+		return errors.New("nil header is invalid")
+	}
+	if dah.IsZero() {
+		return errors.New("must have at least one row and column roots")
+	}
+	if len(dah.ColumnRoots) != len(dah.RowsRoots) {
+		return fmt.Errorf(
+			"unequal number of row and column roots: row %d col %d",
+			len(dah.RowsRoots),
+			len(dah.ColumnRoots),
+		)
+	}
+	if len(dah.hash) == 0 {
+		dah.Hash()
+	}
+
+	return nil
+}
+
+func (dah *DataAvailabilityHeader) IsZero() bool {
+	if dah == nil {
+		return true
+	}
+	return len(dah.ColumnRoots) == 0 || len(dah.RowsRoots) == 0
+}
+
+// MinDataAvailabilityHeader returns a hard coded copy of a data availability
+// header from empty block data
+func MinDataAvailabilityHeader() *DataAvailabilityHeader {
```

Comment:
Can you also update the comment slightly: https://github.com/lazyledger/lazyledger-core/blob/d865cc7c917114f36b1d56e727c1a35c38d68c74/types/block.go#L312-L316 
(hash(empty) can be still misunderstood as hash(nil))

Reply:
this code is not really reachable, as `ComputeShares` will always return at least 1 share. While not the most explicit, `ComputeShares` returning at least a single share is also currently tested in [TestEmptyBlockData](https://github.com/lazyledger/lazyledger-core/blob/5c39da76bbc461074c97ed29585f1c549b6cca5f/types/block_test.go#L237). Therefore, I decided to remove this code entirely https://github.com/lazyledger/lazyledger-core/pull/312/commits/8acce6bab542afb6149a28e62b4fc9764585943c

happy to reverse that decision, though, if anyone thinks otherwise

Reply:
also added better docs to `ComputeShares` to reflect that it always returns at least a single share https://github.com/lazyledger/lazyledger-core/pull/312/commits/304537fb9e2c141e4f2e78c1196d979e617d4ee4